File: 250601/boj_11650.py
# 11650번(좌표정렬)
# 메모리: 46504kb
# 시간: 2928ms

# 2차원 평면 위의 점 N개가 주어진다. 
# 좌표를 x좌표가 증가하는 순으로, x좌표가 같으면 y좌표가 증가하는 순서로 정렬해라.

n = int(input())
points = []

for i in range(n):
    x,y = map(int,input().split())
    points.append((x,y))

points.sort()
for x, y in points:
    print(x, y)

# 파이썬은 sort()를 이용해서 튜플을 포함한 리스트,문자열,숫자 등 
# 어떤 자료형이든 기본 비교가 가능하면 자동 정렬이 됨.

File: 250601/boj_11650.java
// 11650번 자바 버전 풀이

import java.util.*;

public class boj_11650 {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        int[][] points = new int[n][2]; // 2차원 배열로 좌표 저장

        for (int i = 0; i < n; i++){
            points[i][0] = sc.nextInt(); // x좌표
            points[i][1] = sc.nextInt(); // y좌표
        }

        // 정렬: x 오름차순, x가 같으면 y 오름차순
        Arrays.sort(points, new Comparator<int[]>() {
            public int compare(int[] a, int[] b) {
                if (a[0] == b[0]) {
                    return a[1] - b[1]; // y 기준 정렬 
                }
                return a[0] - b[0]; // x 기준 정렬
            }
        });

        //출력
        for (int[] point : points) {
            System.out.println(point[0] + " " + point[1]);
        }

        sc.close();
    }
}